sudoku.js

```diff
@@ -1,24 +1,152 @@
 "use strict"
+const colors = {
+  black: "\x1b[30m",
+  gray: "\x1b[90m",
+  red: "\x1b[31m",
+  green: "\x1b[32m",
+  yellow: "\x1b[33m",
+  blue: "\x1b[34m",
+  magenta: "\x1b[35m",
+  cyan: "\x1b[36m",
+  white: "\x1b[37m",
+  crimson: "\x1b[38m",
+  default: "\x1b[0m",
+ }
 
 class Sudoku {
-  constructor(board_string) {}
+  constructor(board_string) {
+    this.gameBoard =[];
+    this.predefineCoordinate = [];
+    let counter = 0;
+    for(let i=0; i<9; i++){
+      this.gameBoard[i] = [];
+      for(let j=0; j<9; j++){
+        this.gameBoard[i][j] = board_string[counter];
+        if(board_string[counter] !== "0"){
+          this.predefineCoordinate.push(""+i+j);
```

Comment:
akan lebih baik jika disimpan dalam bentuk array
```suggestion
          this.predefineCoordinate.push([i, j]);
```